src/query/users/src/user_mgr.rs

```diff
@@ -43,7 +48,7 @@ impl UserApiProvider {
                     UserInfo::new_no_auth(&user.username, &user.hostname)
                 }
             };
-            if user.is_localhost() {
+            if client_ip == "127.0.0.1" || client_ip == "localhost" {
```

Comment:
client_ip could be removed.

Reply:
allow root user login from any client ip?